### PR TITLE
Implement `TokenCreateTransaction`

### LINKED ISSUE #348: Implement `TokenCreateTransaction`
Implement the APIs associated with TokenCreation (aka `TokenCreateTransaction`).

### PR DESCRIPTION
**Pull request by rwalworth**
**Description**:
This PR implements the `TokenCreateTransaction` as well as its associated unit and integration tests.

This contains work from #363 and #371, so those should be reviewed and merged first.

**Related issue(s)**:

Fixes #348 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/main/include/TokenCreateTransaction.h [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+   *         data.
+   */
+  [[nodiscard]] proto::TokenCreateTransactionBody* build() const;
+
+  /**
+   * The publicly visible name of the token. The token name is specified as a string of UTF-8 characters in Unicode.
+   * UTF-8 encoding of this Unicode cannot contain the 0 byte (NUL). The token name is not unique. Maximum of 100
+   * characters.
+   */
+  std::string mName;
```
Comment (deyanzz) [inline sdk/main/include/TokenCreateTransaction.h:435]: In [Java SDK](https://github.com/hashgraph/hedera-sdk-java) implementation for `TokenCreateTransaction`, this filed is named [tokenName](https://github.com/hashgraph/hedera-sdk-java/blob/274012d54d75c5896414b8406df76a637eb9c58a/sdk/src/main/java/com/hedera/hashgraph/sdk/TokenCreateTransaction.java#L76).
I would prefer to use the same naming between the C++ and the Java SDKs as the [SDK reference](https://github.com/hashgraph/hedera-sdk-reference/issues) documentation is not well specified yet.

File: sdk/main/include/TokenCreateTransaction.h [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+
+  /**
+   * The publicly visible token symbol. The token symbol is specified as a string of UTF-8 characters in Unicode. UTF-8
+   * encoding of this Unicode cannot contain the 0 byte (NUL). The token symbol is not unique. Maximum of 100
+   * characters.
+   */
+  std::string mSymbol;
```
Comment (deyanzz) [inline sdk/main/include/TokenCreateTransaction.h:442]: In [Java SDK](https://github.com/hashgraph/hedera-sdk-java) implementation for `TokenCreateTransaction`, this filed is named [tokenSymbol](https://github.com/hashgraph/hedera-sdk-java/blob/274012d54d75c5896414b8406df76a637eb9c58a/sdk/src/main/java/com/hedera/hashgraph/sdk/TokenCreateTransaction.java#L83).
As the [SDK reference](https://github.com/hashgraph/hedera-sdk-reference/issues) documentation is not well specified yet, I would prefer to use the same naming between C++ and Java versions.

File: sdk/main/include/TokenCreateTransaction.h [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+
+  /**
+   * The interval at which the auto-renew account will be charged to extend the token's expiry.
+   */
+  std::chrono::duration<double> mAutoRenewPeriod = DEFAULT_AUTO_RENEW_PERIOD;
+
+  /**
+   * The memo associated with the token (UTF-8 encoding max 100 bytes).
+   */
+  std::string mMemo;
```
Comment (deyanzz) [inline sdk/main/include/TokenCreateTransaction.h:517]: Same here - this filed is named [tokenMemo](https://github.com/hashgraph/hedera-sdk-java/blob/274012d54d75c5896414b8406df76a637eb9c58a/sdk/src/main/java/com/hedera/hashgraph/sdk/TokenCreateTransaction.java#L83) in the [Java SDK](https://github.com/hashgraph/hedera-sdk-java).
It's better to use the same naming convention between the C++ and the Java versions.

**Review by deyanzz — CHANGES_REQUESTED** [review]
After I merged the latest version of `main` branch into `00348-implement-tokencreatetransaction` I got a failing `PR Checks` action with the following errors:
 

> [ RUN      ] TokenCreateTransactionIntegrationTest.CanCreateTokenWithListOfTenCustomFixedFees
/runner/_work/hedera-sdk-cpp/hedera-sdk-cpp/sdk/tests/integration/TokenCreateTransactionIntegrationTests.cc:261: Failure
Expected: txReceipt = TokenCreateTransaction() .setName("ffff") .setSymbol("F") .setTreasuryAccountId(AccountId(2ULL)) .setAdminKey(operatorKey) .setCustomFees(fees) .execute(getTestClient()) .getReceipt(getTestClient()) doesn't throw an exception.
  Actual: it throws Hedera::ReceiptStatusException with description "INVALID_TOKEN_ID_IN_CUSTOM_FEES".
/runner/_work/hedera-sdk-cpp/hedera-sdk-cpp/sdk/tests/integration/TokenCreateTransactionIntegrationTests.cc:272: Failure
Expected: tokenId = txReceipt.getTokenId().value() doesn't throw an exception.
  Actual: it throws std::bad_optional_access with description "bad optional access".
[  FAILED  ] TokenCreateTransactionIntegrationTest.CanCreateTokenWithListOfTenCustomFixedFees (287 ms)
[----------] 1 test from TokenCreateTransactionIntegrationTest (287 ms total)


See https://github.com/hashgraph/hedera-sdk-cpp/actions/runs/5313332113/jobs/9636782737#step:15:1551 for further details

File: sdk/main/include/TokenCreateTransaction.h (hunk 1 above)
Comment (rwalworth) [reply sdk/main/include/TokenCreateTransaction.h]: Done.

File: sdk/main/include/TokenCreateTransaction.h (hunk 3 above)
Comment (rwalworth) [reply sdk/main/include/TokenCreateTransaction.h]: Done.

**Pull request merged**